### PR TITLE
Anti-entropy sync services/checks missing entirely from Consul

### LINKED ISSUE #839: Anti-Entropy does not properly handle case with client state and lost-server state
For a long running client, if the server loses the registration of a service or check via complete cluster data loss, a bug in the anti-entropy prevents that from being reconciled until the client is reloaded or restarted.

### PR DESCRIPTION
**Pull request by ryanuber**
Fixes #839
/cc @armon

### CODE REVIEW

File: command/agent/local.go [hunk 1]
```diff
@@ -316,6 +316,13 @@ func (l *localState) setSyncState() error {
...
 	defer l.Unlock()
 
+	for id, _ := range l.services {
+		// If the local service doesn't exist remotely, then sync it
+		if _, ok := services.Services[id]; !ok {
+			l.serviceStatus[id] = syncStatus{inSync: false}
+		}
+	}
+
 	if services != nil {
```
Comment (armon) [inline command/agent/local.go]: It looks like we guard against services result being nil. Can we move that above the new code and just do something like:

```
if services == nil {
 services = make(map...)
}
```
This way we can avoid the guard logic and nesting. I think the new code here may be susceptible to a nil pointer dereference in those cases. Not that I see how we could get nil back.

**Comment by armon** [comment]
Minor nitpick, but LGTM!

File: command/agent/local.go (hunk 1 above)
Comment (ryanuber) [inline command/agent/local.go]: Ah good catch.

Comment (ryanuber) [inline command/agent/local.go]: Addressed in 60a6da2

**Pull request merged**